File: app/src/main/java/com/example/schedule/data/search/network/ScheduleApi.java
package com.example.schedule.data.search.network;

import com.example.schedule.data.dto.ScheduleBetweenResponse;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface ScheduleApi {
    @GET("./search/?transport_types=suburban")
    Call<ScheduleBetweenResponse> getScheduleByStationCodeAndDate(
            @Query("apikey") String apikey,
            @Query("from") String fromCode,
            @Query("to") String toCode,
            @Query("date") String date,
            @Query("transport_types") String transportTypes,
            @Query("lang") String lang
    );
}

File: app/src/main/java/com/example/schedule/data/search/network/ScheduleApi.kt
package com.example.schedule.data.search.network

import com.example.schedule.data.db.entity.AllScheduleResponse
import com.example.schedule.data.dto.ScheduleResponse
import retrofit2.http.GET
import retrofit2.http.Query

interface ScheduleApi {

    @GET("./search/?")
    suspend fun getScheduleByStationCodeAndDate(
        @Query("apikey") apikey: String = "2c748246-2b5d-45f6-8881-d72f287f28f1",
        @Query("from") fromCode: String,
        @Query("to") toCode: String,
        @Query("date") date: String,
        @Query("transport_types") transportTypes: String,
    ): ScheduleResponse

    @GET("./stations_list/?apikey=2c748246-2b5d-45f6-8881-d72f287f28f1&lang=ru_RU&format=json")
    suspend fun getAllStations(): AllScheduleResponse

}